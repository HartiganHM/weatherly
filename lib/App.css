.App {
	background-image: radial-gradient(farthest-corner at 2rem 2rem, #42AAf4 0%, #0A2963 100%);
  text-align: center;
  margin: auto;
  box-sizing: border-box;
}
