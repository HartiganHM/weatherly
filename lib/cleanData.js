function cleanData(data) {
  const sevenDayData = data.forecast.simpleforecast.forecastday.map(day => {
    return {
      day: day.date.weekday,
      icon: weatherIcon(day.icon),
      high: day.high.fahrenheit + "°",
      low: day.low.fahrenheit + "°"
    };
  });
  const tenHourData = data.hourly_forecast.map(obj => {
    let temp = obj.temp.english;
    let icon = weatherIcon(obj.icon);
    let hour = obj.FCTTIME.hour;
    let minute = obj.FCTTIME.min;
    let meridiem = " AM";

    if (hour === "0") {
      hour = 12;
    }

    if (hour > 12) {
      hour = hour - 12;
      meridiem = " PM";
    }
    return {
      hour: hour + ":" + minute + meridiem,
      icon,
      temp: temp + "°"
    };
  });

  // set variable data.forecast.simpleforecast.forecastday[0]
  const currentWeatherData = {
    city: data.current_observation.display_location.full,
    icon: weatherIcon(data.hourly_forecast[0].icon),
    condition: data.current_observation.weather,
    day: data.forecast.txt_forecast.forecastday[0].title,
    date:
      data.forecast.simpleforecast.forecastday[0].date.monthname +
      " " +
      data.forecast.simpleforecast.forecastday[0].date.day +
      ", " +
      data.forecast.simpleforecast.forecastday[0].date.year,
    tempF: data.current_observation.temp_f,
    tempC: data.current_observation.temp_c,
    tempHi: data.forecast.simpleforecast.forecastday[0].high.fahrenheit,
    tempLo: data.forecast.simpleforecast.forecastday[0].low.fahrenheit,
    summary: data.forecast.txt_forecast.forecastday[0].fcttext
  };

  return { currentWeatherData, tenHourData, sevenDayData };
}

function weatherIcon(weather) {
  return "styles/icons/white/svg/" + weather + ".svg";
}

export default cleanData;
