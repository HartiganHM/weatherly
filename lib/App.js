import React from "react";
import Welcome from "./Welcome";
import Search from "./Search";
import CurrentWeather from "./CurrentWeather";
import TenHour from "./TenHour";
import SevenDay from "./SevenDay";
import denverData from "./denverData";
import key from "./key.js";
import cleanData from "./cleanData";
import "../styles/App.css";

export default class App extends React.Component {
  constructor() {
    super();
    this.state = {};
    this.fetchData = this.fetchData.bind(this);
  }

  testInput(value) {
    if (isNaN(parseInt(value))) {
      return `${value[1]}/${value[0]}`;
    } else {
      return value;
    }
  }

  fetchData(location) {
    this.testInput(location);
    fetch(
      `http://api.wunderground.com/api/${key}/forecast10day/hourly/conditions/q/${location[1]}/${location[0]}.json`
    )
      .then(response => response.json())
      .then(data => cleanData(data))
      .then(obj => this.setState(obj))
      .catch(error => {
        console.log(error);
      });
  }

  componentWillMount(location) {
    if (location === undefined) {
      location = JSON.parse(localStorage.getItem("location"));
    }

    if (location !== null) {
      this.fetchData(location);
    }
  }

  render() {
    if (this.state.currentWeatherData) {
      let { currentWeatherData, tenHourData, sevenDayData } = this.state;
      return (
        <div>
          <div className="App">
            <Search fetch={this.fetchData} />
            <CurrentWeather data={currentWeatherData} />
            <TenHour data={tenHourData} />
            <SevenDay data={sevenDayData} />
          </div>
        </div>
      );
    } else if (!JSON.parse(localStorage.getItem("location"))) {
      return <Welcome fetch={this.fetchData} />;
    }

    return null;
  }
}
