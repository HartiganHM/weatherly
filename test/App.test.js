import React from "react";
import { shallow, mount } from "enzyme";
import App from "../lib/App";
import cleanData from "../lib/cleanData";
import denverData from "../lib/denverData";



global.localStorage = {
  getItem(keyword) {
    if (!global.localStorage[keyword]) {
      return null;
    }
    return JSON.stringify(global.localStorage[keyword]);
  },
  setItem(keyword, value) {
    global.localStorage[keyword] = value;
  }
};

describe("App", () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />);
  });

  it("App should exist", () => {
    expect(app).toBeDefined();
  });

  it("App should mount <Welcome />", () => {
    expect(app.find("Welcome")).toBeDefined();
    expect(app.find("Welcome").length).toEqual(1);
  });

  it("should be a function", () => {
    expect(typeof app.state).toEqual("function");
  });

  it("should render the weather page when a city is specified", () => {
    app.setState(cleanData(denverData))

    expect(app.find("Search")).toBeDefined();
    expect(app.find("Search").length).toEqual(1);

    expect(app.find("CurrentWeather")).toBeDefined();
    expect(app.find("CurrentWeather").length).toEqual(1);

    expect(app.find("TenHour")).toBeDefined();
    expect(app.find("TenHour").length).toEqual(1);

    expect(app.find("SevenDay")).toBeDefined();
    expect(app.find("SevenDay").length).toEqual(1);
  });
});
