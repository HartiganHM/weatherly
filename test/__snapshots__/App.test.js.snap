// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should mount <Welcome /> 1`] = `[Function]`;

exports[`App should render the weather page when a city is specified 1`] = `[Function]`;
